### ISSUE TITLE
JavaNCSS count switch statements inconsistently

### ISSUE THREAD

**Issue by nrmancuso** [checkstyle/checkstyle#15437]
I have read check documentation:https://checkstyle.org/checks/metrics/javancss.html#JavaNCSS
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="JavaNCSS">
            <property name="methodMaximum" value="1"/>
        </module>
    </module>
</module>

➜  src cat Test.java 
public class Test {
    void m0(String s) {
        switch (s) { // violation, NCSS is 4
            case "a" -> System.out.println(s);
            default -> System.out.println(s);
        }
    }

    void m1(String s) { // violation, NCSS is 6
        switch (s) {
            case "a": System.out.println(s);
            default: System.out.println(s);
        }
    }
}

➜  src javac Test.java 

➜  src java -jar checkstyle-10.17.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:2:5: NCSS for this method is 4 (max allowed is 1). [JavaNCSS]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:9:5: NCSS for this method is 6 (max allowed is 1). [JavaNCSS]
Audit done.
Checkstyle ends with 2 errors.
```

----------------
I would expect both methods to have the same NCSS.

**Comment by rnveach** [comment]
According to official util ( https://github.com/checkstyle/checkstyle/issues/15080 ) , method m1's count seems correct. Referenced issue should also eventually expand on what is counted.

Printing the counts from our Check results in (line numbers off by 10):
````
CLASS_DEF[11x0]
METHOD_DEF[12x4]
switch[13x8]
case[14x12]
default[15x12]
METHOD_DEF[19x4]
switch[20x8]
case[21x12]
EXPR[21x40]
default[22x12]
EXPR[22x39]
````

It seems the count mis-match is because of 2 `EXPR` tokens the other method doesn't not "have".

> I would expect both methods to have the same NCSS.

Do we know if `m0` should officially have expressions in the `->`?

**Comment by rnveach** [comment]
Example using `{}` with `->`.

````
$ cat TestClass.java
public class Test { // violation, NCSS is 17
    void m0(String s) {
        switch (s) { // violation, NCSS is 4
            case "a" -> System.out.println(s);
            default -> System.out.println(s);
        }
    }

    void m0a(String s) {
        switch (s) { // violation, NCSS is 6
            case "a" -> { System.out.println(s); }
            default -> { System.out.println(s); }
        }
    }

    void m1(String s) { // violation, NCSS is 6
        switch (s) {
            case "a": System.out.println(s);
            default: System.out.println(s);
        }
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <property name="haltOnException" value="false"/>

    <module name="TreeWalker">
        <module name="JavaNCSS">
            <property name="classMaximum" value="0"/>
            <property name="methodMaximum" value="0"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-10.17.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[WARN] TestClass.java:1:1: NCSS for this class is 17 (max allowed is 0). [JavaNCSS]
[WARN] TestClass.java:2:5: NCSS for this method is 4 (max allowed is 0). [JavaNCSS]
[WARN] TestClass.java:9:5: NCSS for this method is 6 (max allowed is 0). [JavaNCSS]
[WARN] TestClass.java:16:5: NCSS for this method is 6 (max allowed is 0). [JavaNCSS]
Audit done.
````

This could just be a misunderstanding of `->` with and without `{}`. I don't know the NCSS well enough if it should count ¿statements? without expressions.

**Comment by rnveach** [comment]
Partial tree:
````
m0:
        |      |  |  |--LITERAL_CASE -> case [4:12]
        |      |  |  |  `--EXPR -> EXPR [4:17]
        |      |  |  |      `--STRING_LITERAL -> "a" [4:17]
        |      |  |  |--LAMBDA -> -> [4:21]
        |      |  |  |--EXPR -> EXPR [4:42]
m0a:
        |      |  |  |--LITERAL_CASE -> case [11:12]
        |      |  |  |  `--EXPR -> EXPR [11:17]
        |      |  |  |      `--STRING_LITERAL -> "a" [11:17]
        |      |  |  |--LAMBDA -> -> [11:21]
        |      |  |  `--SLIST -> { [11:24]
        |      |  |      |--EXPR -> EXPR [11:44]
m1:
        |      |  |  |--LITERAL_CASE -> case [18:12]
        |      |  |  |  |--EXPR -> EXPR [18:17]
        |      |  |  |  |  `--STRING_LITERAL -> "a" [18:17]
        |      |  |  |  `--COLON -> : [18:20]
        |      |  |  `--SLIST -> SLIST [18:40]
        |      |  |      |--EXPR -> EXPR [18:40]
````

There is an expression in the tree for every case. I am leaning to agree this is a valid bug without knowing NCSS that well.
